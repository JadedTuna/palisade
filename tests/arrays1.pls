in {
    high arg: bool;
}
out {
    low arr[3]: bool;
}

// array assignments are done by-value (copy)
x[3] := [false, true, arg];
y[3] := arr;
arr = x;
arr = y;
