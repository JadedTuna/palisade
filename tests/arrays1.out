[1;34mnote: [0mlabel of [1;34marr[0m[[1;34m2[0m] set to [1;93mhigh[0m
  11 | arr [1;34m=[0m x;
       [1;34m~~~~[0m[1;34m^[0m
[1;34mnote: [0mlabel of [1;34marr[0m[[1;34m2[0m] set to [1;93mlow[0m
  12 | arr [1;34m=[0m y;
       [1;34m~~~~[0m[1;34m^[0m
[[1;32m OK [0m] [1;93mlow [0m [1;34marr[0m is [1;93mlow[0m
